### PR TITLE
fix typos in react documentation

### PR DESCRIPTION
**Pull request by garanews**
fix typo in react documentation

### CODE REVIEW

**Review by SSwiniarski — APPROVED** [review]
@garanews, looks good.

File: content/react/concepts/components/components.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ In React, components make up the UI of the application. Components are pieces of
```
Comment (Dusch4593) [inline content/react/concepts/components/components.md:18]: My suggestions:

Typo also found on line # 3. Let's rephrase the `Description` like so:

```
Description: 'Components are pieces of code that are reusable, independent, and make up the UI of a React application.'
```

On line # 14, let's make some edits including emphasizing the first mention of "components" by **bolding** it, like so:

```
In React, **components** are pieces of reusable, independent code that make up the user interface (UI) of the application. Each component comes with one or both of the following:
```

File: content/react/concepts/virtual-dom/virtual-dom.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
...ike a lightweight copy. A virtual DOM object has the same properties as a real DOM object, but it lacks the real thing’s power to directly change what’s on the screen. Manipulating the DOM is slow. Manipulating the virtual DOM is much faster, because nothing gets drawn onscreen. Think of manipulating the virtual DOM as editing a blueprint, as opposed to moving rooms in an actual house. With Virtual DOM, when changes are made to the UI and there is a re-render, they are compared to the relevant branches in the DOM-tree. Then, the changes are applied to the DOM-tree. React Virtual DOM compared with the Real DOM The Virtual DOM is not to be confused with the Shadow DOM, or any other technology or concept. When you render a JSX element, every single virtual DOM object gets updated. This sounds incredibly inefficient, but the cost is insignificant because the virtual DOM can update so quickly. Once the virtual DOM has updated, then React compares the virtual DOM with a virtual DOM snapshot that was taken right before the update. By comparing the new virtual DOM with a pre-update version, React figures out exactly which virtual DOM objects have changed. This process is called “diffing.”'
```
Comment (Dusch4593) [inline content/react/concepts/virtual-dom/virtual-dom.md:3]: Per our [entry template](https://github.com/Codecademy/docs/blob/195d731f45c186e95a7b1effb92d41e1d09afe58/documentation/entry-template.md?plain=1#L3), let's shorten this `Description` to (ideally) 150 characters or less.

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@garanews Finished my review and left some comments/suggestions. Here are some other suggested edits:

Under `lifecycle-methods.md`:
- On line # 3, per our [entry template](https://github.com/Codecademy/docs/blob/195d731f45c186e95a7b1effb92d41e1d09afe58/documentation/entry-template.md?plain=1#L3), let's shorten the `Description` to (ideally) 150 characters or less.
- On line # 189, add a `>` at the beginning (i.e., `> **Note:** ...`).

Under `virtual-dom.md`:
- Change all instances of "Virtual DOM" to "virtual DOM" (unless it is the `Title` or first word in a sentence). Right now there is a mix and [the official React docs](https://reactjs.org/docs/faq-internals.html) refer to it as such.
- On line # 3, emphasize the first mention of "virtual DOM" in **bold**, like so:
  `In React, the **virtual DOM** is a conceptual ...`
- Omit any and all uses of "you" and "your" and rephrase if necessary. We've been trying to make Docs more presentation/declarative in tone without addressing the reader(s) directly with something like "you."

File: content/react/concepts/components/components.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 ---
 Title: 'Components'
-Description: 'Make up the UI of the application. Components are pieces of code that are reuseable and independent.'
+Description: 'Description: 'Components are pieces of code that are reusable, independent, and make up the UI of a React application.'
```
Comment (Dusch4593) [inline content/react/concepts/components/components.md:3]:
```suggestion
Description: 'Components are pieces of code that are reusable, independent, and make up the UI of a React application.'
```

File: content/react/concepts/lifecycle-methods/lifecycle-methods.md [hunk 4]
```diff
@@ -186,11 +186,11 @@ class App extends React.Component {
 }
 ```
 
-**Note:** `getDerivedStateFromError()` and `componentDidCatch()` are only able to work with errors strictly related to their child components. They _are not_ compatible with errors related to event handlers.
+>**Note:** `getDerivedStateFromError()` and `componentDidCatch()` are only able to work with errors strictly related to their child components. They _are not_ compatible with errors related to event handlers.
```
Comment (Dusch4593) [inline content/react/concepts/lifecycle-methods/lifecycle-methods.md:189]:
```suggestion
> **Note:** `getDerivedStateFromError()` and `componentDidCatch()` are only able to work with errors strictly related to their child components. They are not compatible with errors related to event handlers.
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
A few last comments/suggestions from me:

Under `lifecycle-methods.md`:
- On line # 3, per our [entry template](https://github.com/Codecademy/docs/blob/195d731f45c186e95a7b1effb92d41e1d09afe58/documentation/entry-template.md?plain=1#L3), let's shorten the `Description` to (ideally) 150 characters or less.

File: content/react/concepts/lifecycle-methods/lifecycle-methods.md (hunk 4 above)
Comment (garanews) [reply content/react/concepts/lifecycle-methods/lifecycle-methods.md:189]: eheh, about note still waiting for your internal discussion, about change 150 files ;)

File: content/react/concepts/virtual-dom/virtual-dom.md [hunk 5]
```diff
@@ -12,27 +12,27 @@ CatalogContent:
...
-![React Virtual DOM compared with the Real DOM](https://i.imgur.com/dIejeNk.png)
+![React virtual DOM compared with the Real DOM](https://i.imgur.com/dIejeNk.png)
 
-The Virtual DOM is not to be confused with the Shadow DOM, or any other technology or concept.
+The virtual DOM is not to be confused with the Shadow DOM, or any other technology or concept.
 
 ## How It Works
 
-When you render a JSX element, every single virtual DOM object gets updated. This sounds incredibly inefficient, but the cost is insignificant because the virtual DOM can update so quickly. Once the virtual DOM has updated, then React compares the virtual DOM with a virtual DOM snapshot that was taken right before the update.
+During a render of a JSX element, every single virtual DOM object gets updated. This sounds incredibly inefficient, but the cost is insignificant because the virtual DOM can update so quickly. Once the virtual DOM has updated, then React compares the virtual DOM with a virtual DOM snapshot that was taken right before the update.
```
Comment (Dusch4593) [inline content/react/concepts/virtual-dom/virtual-dom.md:27]:
```suggestion
During a render of a [JSX](https://www.codecademy.com/resources/docs/react/jsx) element, every single virtual DOM object gets updated. This sounds incredibly inefficient, but the cost is insignificant because the virtual DOM can update so quickly. Once the virtual DOM has updated, then React compares the virtual DOM with a virtual DOM snapshot that was taken right before the update.
```

File: content/react/concepts/virtual-dom/virtual-dom.md [hunk 6]
```diff
@@ -12,27 +12,27 @@ CatalogContent:
...
 
 Once React knows which virtual DOM objects have changed, then React updates those objects, and only those objects, on the real DOM. For example, if we are making changes to a list item, React would be smart enough to rebuild your one checked-off list item, and leave the rest of your list alone.
 
 This makes a big difference! React can update only the necessary parts of the DOM. React’s reputation for performance comes largely from this innovation.
 
-In summary, here’s what happens when you try to update the DOM in React:
+In summary, here’s what happens trying to update the DOM in React:
 
 1. The entire virtual DOM gets updated.
 2. The virtual DOM gets compared to what it looked like before you updated it. React figures out which objects have changed.
```
Comment (Dusch4593) [inline content/react/concepts/virtual-dom/virtual-dom.md:38]:
```suggestion
2. The virtual DOM gets compared to what it looked like before it was updated. React figures out which objects have changed.
```

Comment (Dusch4593) [reply content/react/concepts/virtual-dom/virtual-dom.md:38]: And on the next line, could we go with this suggestion?

```
3. Only the objects that were changed get updated on the real DOM.
```

File: content/react/concepts/virtual-dom/virtual-dom.md [hunk 7]
```diff
@@ -12,27 +12,27 @@ CatalogContent:
...
 
 ## How It Works
 
-When you render a JSX element, every single virtual DOM object gets updated. This sounds incredibly inefficient, but the cost is insignificant because the virtual DOM can update so quickly. Once the virtual DOM has updated, then React compares the virtual DOM with a virtual DOM snapshot that was taken right before the update.
+During a render of a JSX element, every single virtual DOM object gets updated. This sounds incredibly inefficient, but the cost is insignificant because the virtual DOM can update so quickly. Once the virtual DOM has updated, then React compares the virtual DOM with a virtual DOM snapshot that was taken right before the update.
 
 By comparing the new virtual DOM with a pre-update version, React figures out exactly which virtual DOM objects have changed. This process is called “diffing.”
 
 Once React knows which virtual DOM objects have changed, then React updates those objects, and only those objects, on the real DOM. For example, if we are making changes to a list item, React would be smart enough to rebuild your one checked-off list item, and leave the rest of your list alone.
```
Comment (Dusch4593) [inline content/react/concepts/virtual-dom/virtual-dom.md:31]:
```suggestion
Once React knows which virtual DOM objects have changed, then React updates those objects, and only those objects, on the real DOM. For example, if changes were made to a list item, React would know to rebuild the one checked-off list item, and leave the rest of the list alone.
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@garanews Just a few last comments/suggestions from me. There's still a couple "you"/"your" references left to take care of.

**Review by Dusch4593 — APPROVED** [review]
@garanews LGTM! 👍🏻 I made a small edit in c82ba8c378a3e784959bf723f94f458f6d108845 where I shortened the `Description` in `lifecycle-methods.md` and bolded the concept.
